app/controllers/data_requests_controller.rb

```diff
@@ -0,0 +1,71 @@
+class DataRequestsController < ApplicationController
+  def request_extracts
+    make_request(DataRequest.requested_data[:extracts])
+  end
+
+  def request_reductions
+    make_request(DataRequest.requested_data[:reductions])
+  end
+
+  def check_status
+    request = DataRequest.find(params[:request_id])
+    return head 404 if request.nil?
+
+    if request.pending?
+      return head 201
+    end
+
+    if request.processing?
+      return head 202
+    end
+
+    if request.failed?
+      return head 500
+    end
+
+    if request.complete?
+      return head 200
+    end
+
+  end
+
+  def retrieve
+    request = DataRequest.find(params[:request_id])
+    return head 404 if request.nil? or request.url.blank?
+
+    render json: request.url
+  end
+
+  private
+
+  def authorized?
+    workflow.present?
+  end
+
+  def make_request(request_type)
+    request = DataRequest.find_or_initialize_by(
```

Comment:
I know that this deduplication is the same as what Panoptes is also doing, but I'm actually starting to dislike that. Do you feel strongly about it (compared to just creating a new export every time someone requests it)? The downside of deduplicating is that you can't access the old one while the new one is generating, so I think I'd rather just have a complete history of all the DataRequests that were ever generated as individual records.